compute/cloud-client/src/test/java/compute/reservation/CreateReservationFromVmIT.java

```diff
@@ -0,0 +1,108 @@
+
+package compute.reservation;
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth.assertWithMessage;
+
+import com.google.api.gax.rpc.NotFoundException;
+import com.google.cloud.compute.v1.Instance;
+import com.google.cloud.compute.v1.InstancesClient;
+import com.google.cloud.compute.v1.Reservation;
+import com.google.cloud.compute.v1.ReservationsClient;
+import compute.CreateInstance;
+import compute.DeleteInstance;
+import compute.Util;
+import java.io.IOException;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class CreateReservationFromVmIT {
+
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static final String ZONE = "us-east4-c";
+  private static ReservationsClient reservationsClient;
+  private static InstancesClient instancesClient;
+  private static String RESERVATION_NAME;
+  private static String INSTANCE_FOR_RESERVATION;
+  static String javaVersion = System.getProperty("java.version").substring(0, 2);
+
+  // Check if the required environment variables are set.
+  public static void requireEnvVar(String envVarName) {
+    assertWithMessage(String.format("Missing environment variable '%s' ", envVarName))
+        .that(System.getenv(envVarName)).isNotEmpty();
+  }
+
+  @BeforeAll
+  public static void setUp()
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("GOOGLE_CLOUD_PROJECT");
+    reservationsClient = ReservationsClient.create();
+    instancesClient = InstancesClient.create();
+
+    RESERVATION_NAME = "test-reservation-from-vm-" + javaVersion + "-"
+        + UUID.randomUUID().toString().substring(0, 8);
+    INSTANCE_FOR_RESERVATION = "test-instance-for-reserv-" + javaVersion + "-"
+        + UUID.randomUUID().toString().substring(0, 8);
+
+    // Cleanup existing stale resources.
+    Util.cleanUpExistingInstances("test-instance-for-reserv-"  + javaVersion, PROJECT_ID, ZONE);
+    Util.cleanUpExistingReservations("test-reservation-from-vm-"  + javaVersion, PROJECT_ID, ZONE);
+
+    CreateInstance.createInstance(PROJECT_ID, ZONE, INSTANCE_FOR_RESERVATION);
+  }
+
+  @AfterAll
+  public static void cleanup()
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // Delete resources created for testing.
+    DeleteInstance.deleteInstance(PROJECT_ID, ZONE, INSTANCE_FOR_RESERVATION);
+    DeleteReservation.deleteReservation(PROJECT_ID, ZONE, RESERVATION_NAME);
+
+    // Test that reservation is deleted
+    Assertions.assertThrows(
+        NotFoundException.class,
+        () -> GetReservation.getReservation(PROJECT_ID, RESERVATION_NAME, ZONE));
```

Comment:
Can the reservation be deleted as part of the test `testCreateComputeReservationFromVm` itself?

Reply:
Fixed